lint.js

```diff
@@ -68,34 +69,37 @@ const command = async (specFile, cmd) => {
     config.init(cmd);
     const jsonSchema = config.get('jsonSchema');
     const verbose = config.get('quiet') ? 0 : (config.get('verbose') ? 2 : 1);
-    const rulesFiles = config.get('lint:rules');
+    const rulesets = config.get('lint:rules');
     const skip = config.get('lint:skip');
 
-    linter.init();
-    await loader.loadRuleFiles(rulesFiles, { verbose });
+    rules.init({
+        skip
+    });
+    await loader.loadRulesets(rulesets, { verbose });
+    linter.applyRules(rules.getRules());
 
     const spec = await loader.readOrError(
         specFile,
         buildLoaderOptions(jsonSchema, verbose),
     );
 
     validator.validate(spec, buildValidatorOptions(skip, verbose), (err, _options) => {
-        const { context, lintResults } = _options;
+        const { context, warnings } = _options || err.options;
 
-        if (err) {
+        if (err && typeof _options !== 'undefined') {
```

Comment:
There seems to be an issue here, but I haven't found the exact solution yet.  It seems that `_options` is always undefined, so the new conditional check for `_options !== 'undefined'` hides validation errors (e.g., a missing reference is reported as "Specification is valid, with 0 lint errors").  However, removing this check doesn't work either because it seems that `err` is always set for both validation errors and lint warnings, so lint warnings would end up being reported as "Specification schema is invalid" without the new condition, which isn't correct either.  It needs to somehow determine whether `err` represents a validation error vs. just lint warnings.

Reply:
I'll add this to my research in https://github.com/Mermade/oas-kit/issues/137. Feels to me we are doing the reject wrong after the change to `call-me-maybe`.

Reply:
It looks like oas-validator treats lint warnings as errors (`oas-validator/index.js`):

`should (options.warnings.length).be.equal(0,'There were ${options.warnings.length} lint rule violations${options.warnings.length > options.lintLimit ? ', showing first ${options.lintLimit}' : ''}');`

So, either oas-validator should not do that, or the code above needs to somehow distinguish between "real" errors vs. lint warnings.

Reply:
I am leaning towards the last bit, but I can't find a way to do that without a different behavior of `oas-validator`. As I mentioned in that issue.

Reply:
@chris-branch  

I've changed it to:

```javascript
if (err && err.options.valid === false) {
```

That will do the trick to properly distinct between validation errors or linting errors.